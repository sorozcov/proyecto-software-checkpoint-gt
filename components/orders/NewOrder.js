import React from 'react';
import { connect } from 'react-redux';
import { reduxForm, Field } from 'redux-form';
import { StyleSheet, View, KeyboardAvoidingView,Text } from 'react-native';
import { withTheme, Button } from 'react-native-paper';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import MyTextInput from '../textInput';

import * as actions from '../../src/actions/orders';


function NewOrder({ theme, navigation, dirty, valid, handleSubmit, next }){

    const { roundness } = theme

    const proceed = values => {
        if (values.name && values.table) {
            next(navigation, values)
        }
        else{
            changeError(true)
        }
    }

    return(
        <KeyboardAvoidingView
            behavior={Platform.OS == "ios" ? "padding" : "height"}
            style={styles.container}
        >
            <View style={styles.container}>
                <View style={styles.formContainer}>
                    <View style={{alignSelf:"center"}}>
                        <MaterialCommunityIcons name="food" color={'black'} size={125}
                        style={{ marginTop: 0,paddingBottom:0,marginBottom:0 }} title="Hola"/>
                        <Text style={{ fontSize: 22,fontFamily:'dosis-bold',paddingBottom:10,paddingTop:0  }}>NUEVO PEDIDO </Text>
                    </View>
                    <Field name={'table'} component={MyTextInput} label='Mesa' placeholder='No. de Mesa' keyboardType='numeric'/>
                    <Field name={'name'} component={MyTextInput} label='Nombre' placeholder='Ingresa un nombre' returnKeyType='done'/>
                    <View style={styles.buttonContainer}>
                        <Button
                            disabled={!(dirty && valid)}
                            theme={roundness}
                            color={'#000000'}
                            icon={"arrow-right-bold"}
                            height={50}
                            mode="contained"
                            labelStyle={{
                                fontFamily: "dosis-bold",
                                fontSize: 15,
                            }}
                            style={styles.button}
                            onPress={handleSubmit(proceed)}>
                            {'CONTINUAR'}
                        </Button>
                    </View>
                </View>
            </View>
        </KeyboardAvoidingView>
    )
}


const styles = StyleSheet.create({
    button: {
        fontFamily: 'dosis',
        marginLeft: '5%',
        marginRight: '5%',
        justifyContent: 'center',
    },
    buttonContainer: {
        marginTop:'4%',
        marginBottom:'10%',
    },
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#fff',
        fontFamily: 'dosis-regular',
    },
    formContainer: {
        flex: 1,
        justifyContent: 'center',
        paddingTop: 30,
    },
});


export default connect(
    undefined,
    dispatch => ({
        next(navigation, values) {
            navigation.navigate('ProductSelect');
            dispatch(actions.activateOrder(values))
        },
    }),
)(reduxForm({
    form: 'newOrder',
    enableReinitialize : true,
    validate: (values) => {
        const errors = {};

        errors.name = !values.name ? 'Este campo es obligatorio' : undefined;
        errors.table = !values.table ? 'Este campo es obligatorio' : undefined;
        
        return errors;
    }
})(withTheme(NewOrder)));