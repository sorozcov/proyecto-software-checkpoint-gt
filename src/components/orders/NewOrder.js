import React from 'react';
import { KeyboardAvoidingView, StyleSheet, Text, View } from 'react-native';
import { Button, withTheme } from 'react-native-paper';
import MaterialCommunityIcons from 'react-native-vector-icons/MaterialCommunityIcons';
import { connect } from 'react-redux';
import { Field, reduxForm } from 'redux-form';
import * as actions from '../../logic/actions/orders';
import MyTextInput from '../general/textInput';



function NewOrder({ theme, navigation, dirty, valid, handleSubmit, next }){

    console.log("NEW ORDER SCREEN");

    const { roundness } = theme

    const proceed = values => {
        if (values.name && values.table) {
            next(navigation, values);
            values.name = null;
            values.table = null;
        }
        else{
            changeError(true)
        }
    }

    return(
        <KeyboardAvoidingView
            behavior={Platform.OS == "ios" ? "padding" : "height"}
            style={styles.container}
        >
            <View style={styles.container}>
                <View style={styles.formContainer}>
                    <View style={{alignSelf:"center"}}>
                        <MaterialCommunityIcons name="food" color={'black'} size={125}
                        style={{ marginTop: 0,paddingBottom:0,marginBottom:0 }} title="Hola"/>
                        <Text style={{ fontSize: 22,fontFamily:'dosis-bold',paddingBottom:10,paddingTop:0  }}>NUEVO PEDIDO </Text>
                    </View>
                    <Field name={'table'} component={MyTextInput} label='Mesa' placeholder='No. de Mesa' keyboardType='numeric'/>
                    <Field name={'name'} component={MyTextInput} label='Nombre' placeholder='Ingresa un nombre' returnKeyType='done'/>
                    <View style={styles.buttonContainer}>
                        <Button
                            disabled={!(dirty && valid)}
                            theme={roundness}
                            color={'#000000'}
                            icon={"arrow-right-bold"}
                            height={50}
                            mode="contained"
                            labelStyle={{
                                fontFamily: "dosis-bold",
                                fontSize: 15,
                            }}
                            style={styles.button}
                            onPress={handleSubmit(proceed)}>
                            {'CONTINUAR'}
                        </Button>
                    </View>
                </View>
            </View>
        </KeyboardAvoidingView>
    )
}


const styles = StyleSheet.create({
    button: {
        fontFamily: 'dosis',
        marginLeft: '5%',
        marginRight: '5%',
        justifyContent: 'center',
    },
    buttonContainer: {
        marginTop:'4%',
        marginBottom:'10%',
    },
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#fff',
        fontFamily: 'dosis-regular',
    },
    formContainer: {
        flex: 1,
        justifyContent: 'center',
        paddingTop: 30,
    },
});


export default connect(
    undefined,
    dispatch => ({
        next(navigation, values) {
            navigation.navigate('ProductSelect');
            dispatch(actions.activateOrder(values));
        },
    }),
)(reduxForm({
    form: 'newOrder',
    enableReinitialize : true,
    validate: (values) => {
        const errors = {};

        errors.name = !values.name ? 'Este campo es obligatorio' : undefined;
        errors.table = !values.table ? 'Este campo es obligatorio' : undefined;
        
        return errors;
    }
})(withTheme(NewOrder)));