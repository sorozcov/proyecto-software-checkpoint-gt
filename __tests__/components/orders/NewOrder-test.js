import React from 'react';
import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';
import { render } from 'react-native-testing-library';

import NewOrder from '../../../src/components/orders/NewOrder';


const mockStore = configureStore([]);
const store = mockStore({
})

describe("Snapshot", () => {
    it('New Order snapshot', () => {
        const component = render(
            <Provider store={store}><NewOrder /></Provider>
        );
    
        let tree = component.toJSON();
    
        expect(tree).toMatchSnapshot();
    })
    
})
