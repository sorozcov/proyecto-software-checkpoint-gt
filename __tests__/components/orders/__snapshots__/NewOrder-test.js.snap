// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot New Order snapshot 1`] = `
<View
  onLayout={[Function]}
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "column",
        "fontFamily": "dosis-regular",
      },
      Object {
        "paddingBottom": 0,
      },
    ]
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "flexDirection": "column",
        "fontFamily": "dosis-regular",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "justifyContent": "center",
          "paddingTop": 30,
        }
      }
    >
      <View
        style={
          Object {
            "alignSelf": "center",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "fontFamily": "dosis-bold",
              "fontSize": 22,
              "paddingBottom": 10,
              "paddingTop": 0,
            }
          }
        >
          NUEVO PEDIDO 
        </Text>
      </View>
      <View>
        <View
          style={
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 64,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Mesa
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Mesa
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 1,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Mesa
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                keyboardType="numeric"
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "textAlign": "left",
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlignVertical": "center",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View>
        <View
          style={
            Object {
              "paddingLeft": 20,
              "paddingRight": 20,
              "paddingTop": 10,
            }
          }
        >
          <View>
            <View
              pointerEvents="none"
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 6,
                  },
                  Object {
                    "backgroundColor": "#f6f6f6",
                    "borderColor": "rgba(0, 0, 0, 0.54)",
                    "borderRadius": 4,
                    "borderWidth": 1,
                  },
                ]
              }
            />
            <View
              style={
                Object {
                  "minHeight": 64,
                  "paddingBottom": 0,
                  "paddingTop": 8,
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": Array [
                      Object {
                        "translateX": 0,
                      },
                    ],
                  }
                }
              >
                <View
                  pointerEvents="none"
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "bottom": 0,
                      "height": 2,
                      "left": 10,
                      "opacity": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 6,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                      ],
                      "width": 8,
                    }
                  }
                />
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "backgroundColor": "#f6f6f6",
                      "color": "transparent",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 18,
                      "opacity": 0,
                      "paddingHorizontal": 0,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 37,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                        Object {
                          "scaleY": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Nombre
                </Text>
                <Text
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "#6200ee",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 0,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Nombre
                </Text>
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, 0.54)",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "left": 0,
                      "opacity": 1,
                      "paddingHorizontal": 14,
                      "position": "absolute",
                      "textAlign": "left",
                      "top": 36,
                      "transform": Array [
                        Object {
                          "translateX": 0,
                        },
                        Object {
                          "translateY": 0,
                        },
                        Object {
                          "scale": 1,
                        },
                      ],
                      "writingDirection": "ltr",
                    }
                  }
                >
                  Nombre
                </Text>
              </View>
              <TextInput
                allowFontScaling={true}
                editable={true}
                multiline={false}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder=""
                placeholderTextColor="rgba(0, 0, 0, 0.54)"
                rejectResponderTermination={true}
                selectionColor="#6200ee"
                style={
                  Array [
                    Object {
                      "flexGrow": 1,
                      "margin": 0,
                      "paddingHorizontal": 14,
                      "textAlign": "left",
                      "zIndex": 1,
                    },
                    Object {
                      "height": 56,
                    },
                    Object {
                      "paddingBottom": 0,
                      "paddingTop": 0,
                    },
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontSize": 16,
                      "fontWeight": undefined,
                      "textAlignVertical": "center",
                    },
                  ]
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "marginBottom": "10%",
            "marginTop": "4%",
          }
        }
      >
        <View
          height={50}
          style={
            Object {
              "backgroundColor": "rgba(0, 0, 0, 0.12)",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 0,
              "fontFamily": "dosis",
              "justifyContent": "center",
              "marginLeft": "5%",
              "marginRight": "5%",
              "minWidth": 64,
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityStates={
              Array [
                "disabled",
              ]
            }
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <View
                style={
                  Object {
                    "marginLeft": 12,
                    "marginRight": -4,
                    "width": 16,
                  }
                }
              >
                <Text />
              </View>
              <Text
                numberOfLines={1}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      undefined,
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "rgba(0, 0, 0, 0.32)",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "dosis-bold",
                        "fontSize": 15,
                      },
                    ],
                  ]
                }
              >
                CONTINUAR
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
